---
title: "AF"
author: "Malvika Tejura"
date: "2024-02-24"
output: html_document
---

```{r}
library(dplyr)
library(tidyr)

## for this script you will need to add in paths and file names where necessary, redundancy is added to create intermediate files incase of crashes

#read csv with parsed information from annovar annotated file 

clinvar23_no_AF_fltr = read.csv('parsed file from df6')

#replace variants with no gnomad genome or exome frequencies with '0'

clinvar23_no_AF_fltr_1 = clinvar23_no_AF_fltr %>% mutate(across(c(genome_AF, exome_AF), ~ ifelse(genome_AF == '.' & exome_AF == '.', 0, .)))

#change datatype of the AF column

clinvar23_no_AF_fltr_1$exome_AF <- as.numeric(clinvar23_no_AF_fltr_1$exome_AF)

clinvar23_no_AF_fltr_1$genome_AF <- as.numeric(clinvar23_no_AF_fltr_1$genome_AF)

#filter on variants that have less than 0.01 allele frequency

clinvar23_no_AF_fltr_2 = clinvar23_no_AF_fltr_1 %>% filter(case_when(exome_AF == '.' ~ genome_AF < 0.01, TRUE ~ exome_AF < 0.01))

#select relevant columns 

clinvar23_no_AF_fltr_3 = subset(clinvar23_no_AF_fltr_2, select = c('Chr','Start','End','Ref','Alt','REVEL','BayesDel_noAF_score','genome_AF','exome_AF','Otherinfo11','Stars','Gene','GeneID','Significance','clnvc','Variant_type'))

#write csv

write.csv(clinvar23_no_AF_fltr_3, "AF filtered csv", row.names=FALSE)
```

```{r}

#get rid of duplicated gene names and concatenate into one gene 

clinvar23_gene_dupe = read.csv('AF filtered csv')

gene_only = clinvar23_gene_dupe %>% distinct(Gene, .keep_all = TRUE)

gene_only$duplicate = duplicated(gene_only$GeneID)

gd = filter(gene_only, gene_only$duplicate == TRUE)

#duplicated genes are GBA --> GBA1, SPATA5 --> AFG2A , DDX58 --> RIGI, FAM189A1 --> ENTREP2, SPATA5L1 --> AFG2B, KIAA1109 --> BLTP1

clinvar23_gene_dupe$Gene_nodup = clinvar23_gene_dupe$Gene

clinvar23_gene_dupe = clinvar23_gene_dupe %>% mutate(Gene_nodup = ifelse(Gene_nodup == 'GBA','GBA1',ifelse(Gene_nodup == 'SPATA5','AFG2A',ifelse(Gene_nodup == 'DDX58','RIGI', ifelse(Gene_nodup == 'FAM189A1','ENTREP2',ifelse(Gene_nodup == 'SPATA5L1','AFG2B',ifelse(Gene_nodup == 'KIAA1109','BLTP1',Gene_nodup)))))))

colnames(clinvar23_gene_dupe)[colnames(clinvar23_gene_dupe) == "Gene"] <- "OG_Genename"

colnames(clinvar23_gene_dupe)[colnames(clinvar23_gene_dupe) == "Gene_nodup"] <- "Gene"

clinvar23_gene_dupe_no_na = filter(clinvar23_gene_dupe, clinvar23_gene_dupe$REVEL != '.' | clinvar23_gene_dupe$BayesDel_noAF_score != '.')


write.csv(clinvar23_gene_dupe_no_na, "AF filtered with no gene duplicates", row.names=FALSE)

clinvar23_gene_dupe_no_na = read.csv("AF filtered with no gene duplicates")

#filter for disease relevant genes 

#load disease relevant gencc download
gencc = read.csv("GenCC file") #in github folder

#filter on disease associated genes 
gencc_1 = filter(gencc, gencc$classification_title == 'Definitive' | gencc$classification_title == 'Strong' | gencc$classification_title == 'Moderate')

#keep all unique genes
gencc_2 = gencc_1 %>% distinct(gene_symbol, .keep_all = TRUE)

#load all disease relevant genes into a list 
gencc_list = gencc_2$gene_symbol

clinvar23_gene_dupe_no_na = clinvar23_gene_dupe_no_na %>%
  filter(map_lgl(Gene, ~ any(gencc_list %in% .x)))

write.csv(clinvar23_gene_dupe_no_na, ' Clinvar 2023 dataset', row.names=FALSE)

print(length(unique(clinvar23_gene_dupe_no_na$Gene)))

#VUS 
#load in VUS dataframe
bb = read_csv('VUS csv from dfa')

#subset to make it a smaller dataframe
bb_1 = subset(bb, select = c(Chr, Start, End, Ref, Alt, REVEL, BayesDel_noAF_score, genome_AF, exome_AF, Otherinfo11, Stars, GeneID,Significance, clnvc, Variant_type, Gene))

#duplicated genes are GBA --> GBA1, SPATA5 --> AFG2A , DDX58 --> RIGI, FAM189A1 --> ENTREP2, SPATA5L1 --> AFG2B, KIAA1109 --> BLTP1

bb_1$Gene_nodup = bb$Gene

bb_1= bb_1 %>% mutate(Gene_nodup = ifelse(Gene_nodup == 'GBA','GBA1',ifelse(Gene_nodup == 'SPATA5','AFG2A',ifelse(Gene_nodup == 'DDX58','RIGI', ifelse(Gene_nodup == 'FAM189A1','ENTREP2',ifelse(Gene_nodup == 'SPATA5L1','AFG2B',ifelse(Gene_nodup == 'KIAA1109','BLTP1',Gene_nodup)))))))

colnames(bb_1)[colnames(bb_1) == "Gene"] <- "OG_Genename"

colnames(bb_1)[colnames(bb_1) == "Gene_nodup"] <- "Gene"

write.csv(bb_1, "Clinvar 2023 VUS dataset", row.names=FALSE)

```